File: JCodeConverterLite/SFParameter.cpp
//
//  SFParameter.cpp
//  SpotLightTownTest
//
//  Created by Alessandro Martinelli on 14/10/12.
//  Copyright (c) 2012 Alessandro Martinelli. All rights reserved.
//

#include "SFParameter.h"



SFParameter::SFParameter() {
    this->type=GLOBAL_GENERIC;
}

SFParameter::SFParameter(string name) {
    this->name = name;
}

SFParameter::SFParameter(string name, short type) {
    this->name = name;
    this->type = type;
}

string SFParameter::getName() {
    return name;
}

short SFParameter::getType() {
    return this->type;
}

void SFParameter::setType(short type) {
    this->type = type;
}

File: JCodeConverterLite/SFParameter.java
package shadow.pipeline.parameters;


/**
 * @author Alessandro Martinelli
 */
public class SFParameter extends SFParameteri {

	private String name;
	protected short type=GLOBAL_GENERIC;

	public SFParameter() {
		super();
	}

	public SFParameter(String name, short type) {
		super();
		this.name = name;
		this.type = type;
	}


	public SFParameter(String name) {
		super();
		this.name = name;
	}


	/* (non-Javadoc)
	 * @see shadow.pipeline.SFParameteri#getName()
	 */
	@Override
	public String getName() {
		return name;
	}

	/* (non-Javadoc)
	 * @see shadow.pipeline.SFParameteri#getType()
	 */
	@Override
	public long getType() {
		return type;
	}

	public void setType(short type) {
		this.type = type;
	}

	@Override
	public String toString() {
		return name+"("+type+")";
	}
}